torchvision/prototype/datasets/utils/_resource.py

```diff
@@ -32,7 +32,7 @@ def __init__(
         *,
         file_name: str,
         sha256: Optional[str] = None,
-        preprocess: Optional[Union[Literal["decompress", "extract"], Callable[[pathlib.Path], pathlib.Path]]] = None,
+        preprocess: Optional[Union[Literal["decompress", "extract"], Callable[[pathlib.Path], None]]] = None,
```

Comment:
For my own education, why did we need to specify `None` in the annotation here? I assume `Optional[]` would have been enough - is it just to be more explicit about it?

Reply:
It becomes clearer when we let `black` explode the annotation:

```suggestion
        preprocess: Optional[
            Union[
                Literal["decompress", "extract"],
                Callable[[pathlib.Path], None],
            ]
        ] = None,
```

I only changed the return type of the callable from `pathlib.Path` to `None` since we refactored the loading logic that the return is no longer needed.

Still, in general you are right. [`Optional[Foo]` is equivalent to `Union[None, Foo]`](https://github.com/python/cpython/blob/93fc14933b8605c8df23073574048408df61b538/Lib/typing.py#L690-L693). Plus, [`Optional[Union[Foo, Bar]]` can be flattened to `Union[None, Foo, Bar]`](https://github.com/python/cpython/blob/93fc14933b8605c8df23073574048408df61b538/Lib/typing.py#L657-L659). It is just my personal preference to be explicit about `Optional`.